hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -2017,3 +2037,48 @@ func (r *NodePoolReconciler) doesCPODecompressAndDecodeConfig(ctx context.Contex
 	_, managesDecompressAndDecodeConfig := supportutil.ImageLabels(controlPlaneOperatorImageMetadata)[controlPlaneOperatorManagesDecompressAndDecodeConfig]
 	return managesDecompressAndDecodeConfig, nil
 }
+
+func (r *NodePoolReconciler) allMachinesTerminated(nodePool *hyperv1.NodePool) (bool, error) {
```

Comment:
please drop this, see https://github.com/openshift/hypershift/pull/1826/files#r1007344121

Reply:
I'm assuming this function can stay for now as part of gating that all the Machines are gone per - https://github.com/openshift/hypershift/pull/1826#discussion_r1007374630

Reply:
We don't need anything of this nor the loop above.
All we need is a func deleteMachines that short circuit the workflow until there's not Machines persisted in etcd. e.g listMachines == 0

Reply:
I believe this has been addressed now.